pallets/mnt-token/src/lib.rs

```diff
@@ -7,20 +7,70 @@
 
 use frame_support::{pallet_prelude::*, transactional};
 use frame_system::pallet_prelude::*;
-use minterest_primitives::{Balance, CurrencyId, Price, Rate};
+use minterest_primitives::{Balance, CurrencyId, CurrencyPair, Price, Rate};
 pub use module::*;
-use pallet_traits::{LiquidityPoolsManager, PriceProvider};
+use orml_traits::MultiCurrency;
+use pallet_traits::{ControllerAPI, LiquidityPoolsManager, PriceProvider};
 use sp_runtime::{
-	traits::{CheckedDiv, CheckedMul, Zero},
-	DispatchResult, FixedPointNumber,
+	traits::{CheckedAdd, CheckedDiv, CheckedMul, CheckedSub, Zero},
+	DispatchResult, FixedPointNumber, FixedU128,
 };
-use sp_std::{result, vec::Vec};
+use sp_std::{convert::TryInto, result, vec::Vec};
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
 
 #[cfg(test)]
 mod mock;
 #[cfg(test)]
 mod tests;
 
+pub mod weights;
+pub use weights::WeightInfo;
+
+/// Representation of supply/borrow pool state
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, RuntimeDebug, Eq, PartialEq, Default)]
+pub struct MntState<T: Config> {
+	/// Index that represents MNT tokens that distributes for whole pool.
+	/// There is calculation MNT tokens for each user based on this index.
+	pub index: FixedU128,
+	/// The block number the index was last updated at
+	pub block_number: T::BlockNumber,
```

Comment:
```suggestion
	pub index_updated_at_block: T::BlockNumber,
```
As far as I know updated at is convention for such 'last updated'  stuff.